### DISCUSSION TITLE
Update Azure Key Vault secrets documentation

### DISCUSSION THREAD

**Discussion by carolinasantos0** [arcus-azure/arcus.security#312] [General]
Can you update the documentation with a full example of the latest version of the extension Key Vault provider? 
https://security.arcus-azure.net/features/secrets/consume-from-key-vault

**Comment by stijnmoreels** [comment]
Hi! 🙋‍♂️
If you're looking for the Azure Key Vault extension in the Arcus secret store, than [this link](https://security.arcus-azure.net/features/secret-store/provider/key-vault) will guide you through all the available extensions.

If you're looking to extend the Azure Key Vault secret provider, and customize functionality, then the link you mentioned will provide you with this. What we should do, though, is update that example with the new Azure SDK and use the `TokenCredentials` instead. I'll make an issue for that.

Does this answer your question? @carolinasantos0 
Thx for reaching out! 👍

**Reply by carolinasantos0** [reply]
Yes, thank you.

**Reply by ACA-CDT** [reply]
Hi @stijnmoreels , 
I've upgraded yesterday some code to .NET 6.0 and to the latest version of Arcus.
This line in the documentation on [this link](https://security.arcus-azure.net/features/secret-store/provider/key-vault) is no longer valid in the last version of Arcus and needs to be updated:
![image](https://user-images.githubusercontent.com/86767115/165903743-1c41f321-cf04-46a2-ba17-56319b4d1e8c.png)

I've also tried out the example on this page: https://security.arcus-azure.net/features/secrets/consume-from-key-vault
In the indicated lines no longer work in the latest version of Arcus:

![image](https://user-images.githubusercontent.com/86767115/165904272-469e85c2-93f3-4701-ac4e-d4692fdd63f0.png)

**Comment by ACA-CDT** [comment]
Hi @stijnmoreels , 
I've upgraded yesterday some code to .NET 6.0 and to the latest version of Arcus.
This line in the documentation on [this link](https://security.arcus-azure.net/features/secret-store/provider/key-vault) is no longer valid in the last version of Arcus and needs to be updated:
![image](https://user-images.githubusercontent.com/86767115/165903743-1c41f321-cf04-46a2-ba17-56319b4d1e8c.png)

I've also tried out the example on this page: https://security.arcus-azure.net/features/secrets/consume-from-key-vault
In the indicated lines no longer work in the latest version of Arcus:

![image](https://user-images.githubusercontent.com/86767115/165904272-469e85c2-93f3-4701-ac4e-d4692fdd63f0.png)

**Reply by stijnmoreels** [reply]
@HelenaCoppieters , the extension described in the docs called `AddAzureKeyVaultWithManagedIdentity` has several options that can be used. The 'named' secret provider, is one of them. The `...` means that in this example, it doesn't matter which overload with the `name` you use.
If you want to add a named Azure Key Vault secret provider with managed identity, you can use this:

```csharp
stores.AddAzureKeyVaultWithManagedIdentity(
    "https://my-azure-keyvault",
    cacheConfiguration: null,
    configureOptions: null,
    name: "AzureKeyVault.ManagedIdentity",
    mutateSecretName: null);
```

**Reply by stijnmoreels** [reply]
Does this fix your problem, @HelenaCoppieters ? Or is there still something missing?

**Reply by ACA-CDT** [reply]
@stijnmoreels ,  the named Azure Key Vault secret provider with managed identity works with these parameters (only tested this in .Net Core 3.1 and with AzureFunction v3), thx.

**Reply by stijnmoreels** [reply]
Ok, great! Glad this is resolved for you. Don't hesitate to reach out if anything else is not clear!
